code/game/machinery/scan_gate.dm

```diff
@@ -115,6 +118,21 @@
 				if(istype(I, /obj/item/gun))
 					beep = TRUE
 					break
+		if(SCANGATE_HYGIENE)
+			if(iscarbon(M))
+				var/mob/living/carbon/human/H = M
```

Comment:
you checked if its carbon, not if its human

Reply:
time 4 webedits